src/core/lib/json/json_object_loader.cc

```diff
@@ -109,9 +107,13 @@ void LoadString::LoadInto(const std::string& value, void* dst,
   *static_cast<std::string*>(dst) = value;
 }
 
-void LoadUnprocessedJson::LoadInto(const Json& value, void* dst,
-                                   ErrorList*) const {
-  *static_cast<Json*>(dst) = value;
+void LoadUnprocessedJson::LoadInto(const Json& json, void* dst,
+                                   ErrorList* errors) const {
+  if (json.type() != Json::Type::OBJECT) {
```

Comment:
It's unclear to me that this check should be here... isn't it possible we'd want to load an unprocessed Json array (or other type)